db/db_impl.h

```diff
@@ -561,6 +561,13 @@ class DBImpl : public DB {
 
   const SnapshotList& snapshots() const { return snapshots_; }
 
+  const std::vector<SequenceNumber> CopySnapshots(
```

Comment:
Nit: do we consider to pass in the original std::vector<SequenceNumber> and fill it up, to avoid memory allocation?